classes/orcid/actions/AuthorizeUserData.php

```diff
@@ -0,0 +1,138 @@
+<?php
+
+namespace APP\orcid\actions;
+
+use APP\core\Application;
+use APP\core\Request;
+use APP\facades\Repo;
+use APP\orcid\OrcidManager;
+use Carbon\Carbon;
+
+class AuthorizeUserData
+{
+    public function __construct(
+        private Request $request
+    ) {
+    }
+    public function execute(): void
+    {
+        $context = $this->request->getContext();
+        $httpClient = Application::get()->getHttpClient();
+
+        // API Request: GetOAuth token and ORCID
+        $tokenResponse = $httpClient->request(
```

Comment:
Yes, I think so. I'll add one at each step for the external requests, and from that catch, throw a new, custom error that can be caught from the page handler so it can display a failure message to the user. I think this will be especially important in cases where a user is trying to authorize their own ORCID: if something went wrong and there was no error, that would be bad.